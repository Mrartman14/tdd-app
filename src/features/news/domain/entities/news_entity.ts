interface INewsEntity {
    title: string,
    text: string,
}

interface INewsEntityProps {
    title: string,
    text: string,
}

class NewsEntity implements INewsEntity {
    title: string;
    text: string;

    constructor(props: INewsEntityProps) {
        this.title = props.title;
        this.text = props.text;
    }
}

export {
    NewsEntity,
};