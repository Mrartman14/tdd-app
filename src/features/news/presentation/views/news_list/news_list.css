.news-list .news-card:not(:last-child) {
    margin-bottom: 10px;
}