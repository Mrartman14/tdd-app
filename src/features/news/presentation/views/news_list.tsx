import React from 'react';

import { NewsEntity } from '../../domain/entities/news_entity';
import { NewCard } from '../components/newcard/newcard';


interface NewCardProps {
    data: NewsEntity[];
}

class NewsList extends React.PureComponent<NewCardProps> {
    render() {
        const newsListView = this.props.data.map((item, i) => (
            <NewCard data={item} key={i} />
        ));

        return (
            <div className="news-list">
                {newsListView}
            </div>
        );
    }
}

export {
    NewsList,
};