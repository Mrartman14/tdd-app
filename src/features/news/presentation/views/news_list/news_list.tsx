import React from 'react';

import { NewsEntity } from '../../../domain/entities/news_entity';
import { NewsCard } from '../../components/news_card/news_card';

import './news_list.css';
interface NewsListProps {
    data: NewsEntity[];
}
class NewsList extends React.PureComponent<NewsListProps> {
    render() {
        const newsListView = this.props.data.map((item, i) => (
            <NewsCard data={item} key={i} />
        ));

        return (
            <div className="news-list">
                {newsListView}
            </div>
        );
    }
}

export {
    NewsList,
};