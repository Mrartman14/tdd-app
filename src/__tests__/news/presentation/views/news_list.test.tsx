import React from "react";
import { render, unmountComponentAtNode } from "react-dom";

import { act } from "react-dom/test-utils";

import { NewsEntity } from "../../../../features/news/domain/entities/news_entity";
import { NewsList } from "../../../../features/news/presentation/views/news_list/news_list";

const container: Element = document.createElement('div');
const fakeData: NewsEntity[] = [
    {
        title: 'test title 1',
        text: 'test text 1',
    },
    {
        title: 'test title 2',
        text: 'test text 2',
    },
];

beforeEach(() => {
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
});

describe('should news list component renders well', () => {
    it('should news list makes well snapshot', () => {
        act(() => {
            render(<NewsList data={fakeData} />, container);
        });
    
        expect(container.textContent).toMatchSnapshot();
    });

    it('should renders right count of news cards', () => {
        act(() => {
            render(<NewsList data={fakeData} />, container);
        });

        const newsCards = container.querySelectorAll('.news-card');
        expect(newsCards.length).toBe(fakeData.length);
    });
});
