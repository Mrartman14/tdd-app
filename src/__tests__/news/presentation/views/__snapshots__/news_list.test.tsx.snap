// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should news list component renders well 1`] = `"test title 1test text 1test title 2test text 2"`;

exports[`should news list component renders well should news list makes well snapshot 1`] = `"test title 1test text 1test title 2test text 2"`;
